pkgs/build-support/kernel/modules-closure.sh

```diff
@@ -19,36 +19,55 @@ version=$(cd $kernel/lib/modules && ls -d *)
 echo "kernel version is $version"
 
 # Determine the dependencies of each root module.
-closure=
+mkdir -p $out/lib/modules/"$version"
+touch closure
```

Comment:
Using an array could be nicer here, since it avoids word-splitting "fun":

```
closure=()
[…]
closure+=("$module")
[…]
for module in "${closure[@]}"; do […]
```

Reply:
The reason I used a file here was because the `while read` loop seems to run in a subshell, so code afterwards can't use any variables that are set in the loop. I personally don't have experience with arrays in Bourne-based shells (I use fish where I can) but I would assume the same holds for them. Anyway, thanks for merging. :)

Reply:
Aah, I see. Yeah, you can work around that by making what's piped into `while` run in the subshell instead, as in…

```
while read cmd module args ; do
  […]
done < <(modprobe --config no-config -d $kernel --set-version "$version" --show-depends "$module")
```

but that does make reading it a little more confusing. Alternatively:

```
exec 5< <(modprobe --config no-config -d $kernel --set-version "$version" --show-depends "$module")
while read -u 5 cmd module args ; do
  […]
done
```

but yeah I'd say leave it as is now :)